src/main/java/TicTacToe.java

```diff
@@ -0,0 +1,327 @@
+            if( Character.isDigit(move_str.toCharArray()[0]) ) {
+                move_int = Integer.parseInt(move_str);
+                if( ( move_int <= (TicTacToe.N)*(TicTacToe.N) ) && move_int >= 1 ) {
+                    valid_input = true ;
+                    break ;
+                }
+            }
+
+            if( !valid_input ) {
+                System.out.println("Invalid input");
+                continue ;
+            }
+        }
+        return move_int ;
+    }
+
+}
+
+class TicTacToe
+{
+    protected static final int N = 3 ;
+    private static final int HSPACE = 20 ;
+    protected int[][] board;
+    private static BufferedReader reader =
+        new BufferedReader(new InputStreamReader(System.in)) ;
+
+    class Player
+    {
+        private String name ;
+        private int player_type ;
+        private int player_order ;
+        private MoveMethod move_strategy ;
+
+        public Player(String pname, int type, int order, MoveMethod move_s )
+        {
+            name = pname ;
+            player_type = type ;
+            player_order = order ;
+            move_strategy = move_s ;
+        }
+
+        public String getName() {
+            return name ;
+        }
+
+        public int getPlayerType() {
+            return player_type ;
+        }
+
+        public int getMove() {
+            return move_strategy.move();
+        }
+    }
+
+    private Player player1,player2;
+
+    public Player getplayer1() {
+        return player1 ;
+    }
+
+    public Player getplayer2() {
+        return player2 ;
+    }
+
+    public static String getPosDescription(int pos) {
+        String str = "";
+        if( pos == 5 ) {
+            str = "center" ;
+            return str ;
+        }
+
+        if( (pos-1)/3 == 0 ) {
+            str += "upper " ;
+        }
+        else if( (pos-1)/3 == 1 ) {
+            str += "middle " ;
+        }
+        else
+            str += "lower " ;
+
+        if( (pos-1) % 3 == 0 )
+            str += "left" ;
+        else if( (pos-1)%3 == 1 )
+            str += "middle" ;
+        else
+            str += "right" ;
+
+        return str ;
+    }
+
+    protected static String getUserInput() {
+        String input = "" ;
+        try {
+            input = reader.readLine();
+        } catch (IOException ex) {
+            ex.printStackTrace();
+        }
+        return input ;
+    }
+
+    public TicTacToe()
+    {
+        board = new int[N][N];
+        for( int i = 0 ; i < N ; i ++ ) {
+            for( int j = 0 ; j < N ; j ++ ) {
+                board[i][j] = 0 ;
+            }
+        }
+
+        System.out.println("Enter player name");
+        player1 = new Player(getUserInput(),2,0,new HumanMove(this));
+
+        player2 = new Player("",1,1,new SimpleMoveStrategy(this));
+        System.out.println("\nHuman player " + player1.getName() + " vs Computer Player " + player2.getName() + ":" ) ;
+    }
+
+    public boolean setMove(int move, int p_type)

```

Comment:
p_type is a meaningless variable name. This is trying to say the player type so if need should be playerType